ua/org.eclipse.tips.ide/src/org/eclipse/tips/ide/internal/provider/TipHtml.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2023 ArSysOp
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Nikifor Fedorov (ArSysOp) - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.tips.ide.internal.provider;
+
+import java.util.function.Supplier;
+
+final class TipHtml implements Supplier<String> {
+
+	private final String header;
+	private final String body;
+	private final String footer;
+
+	TipHtml(String header, String body, String footer) {
+		this.header = header;
+		this.body = body;
+		this.footer = footer;
+	}
+
+	TipHtml(String header, String body) {
+		this(header, body, "<br><br><br>"); //$NON-NLS-1$
```

Comment:
When instantiating this class without a footer, there will now be 7 line breaks at the end of the document (2+2 from the formatting in `get()` and these three). As this was not the case before (documents ended with two or three line breaks), I would like to know whether this is intended, i.e., whether there was so from with too few line breaks in the existing code that is fixed in this refactoring, or whether it is "by accident".

Reply:
Thank you for noting this! My idea was to show three new lines in case if footer is empty (like in `ActionTips.java`). Updated this part a bit.

Reply:
I see. Looks good with recent changes, thanks!